### PR TITLE
Add support for WRP to HTTP header conversion

### PR DESCRIPTION
**Pull request by shilpa24balaji**

### CODE REVIEW

File: wrp/header_wrp.go [hunk 1]
```diff
@@ -9,17 +9,23 @@ import (
...
-	TrasactionUuidHeader = "X-Midt-Transaction-Uuid"
-	StatusHeader         = "X-Midt-Status"
-	RDRHeader            = "X-Midt-Request-Delivery-Reponse"
-	HeadersArrHeader     = "X-Midt-Headers"
-	IncludeSpansHeader   = "X-Midt-Include-Spans"
-	SpansHeader          = "X-Midt-Spans"
-	PathHeader           = "X-Midt-Path"
-	SourceHeader         = "X-Midt-Source"
+	MsgTypeHeader         = "X-Midt-Msg-Type"
+	ContentTypeHeader     = "X-Midt-Content-Type"
```
Comment (johnabass) [inline wrp/header_wrp.go]: This should just be the standard Content-Type header.

File: wrp/header_wrp.go [hunk 2]
```diff
@@ -9,17 +9,23 @@ import (
...
+	AcceptHeader          = "X-Midt-Accept"
```
Comment (johnabass) [inline wrp/header_wrp.go]: This should just be the standard Accept header

File: wrp/header_wrp.go [hunk 3]
```diff
@@ -9,17 +9,23 @@ import (
...
+	TransactionUuidHeader = "X-Midt-Transaction-Uuid"
+	StatusHeader          = "X-Midt-Status"
+	RDRHeader             = "X-Midt-Request-Delivery-Reponse"
+	HeadersArrHeader      = "X-Midt-Headers"
+	IncludeSpansHeader    = "X-Midt-Include-Spans"
+	SpansHeader           = "X-Midt-Spans"
+	PathHeader            = "X-Midt-Path"
+	SourceHeader          = "X-Midt-Source"
+
+	ErrInvalidMsgType         = "Invalid Message Type"
```
Comment (johnabass) [inline wrp/header_wrp.go]: Each of these should be constants of type `error`:

    ErrInvalidMsgType = errors.New("Invalid Message Type")

File: wrp/header_wrp.go [hunk 4]
```diff
@@ -57,7 +63,7 @@ func HeaderToWRP(header http.Header) (*Message, error) {
 	if !strings.EqualFold(msgType, "") && StringToMessageType(msgType) != MessageType(-1) {
 		msg.Type = StringToMessageType(msgType)
 	} else {
-		return nil, fmt.Errorf("%s", "Invalid Message Type header string")
+		return nil, fmt.Errorf("%s", ErrInvalidMsgType)
```
Comment (johnabass) [inline wrp/header_wrp.go]: Remove the `fmt.Errorf` here in favor of `error` constants (see above)

File: wrp/header_wrp.go [hunk 5]
```diff
@@ -66,16 +72,16 @@ func HeaderToWRP(header http.Header) (*Message, error) {
 	} else if msg.Type == SimpleRequestResponseMessageType || msg.Type == SimpleEventMessageType ||
 		msg.Type == CreateMessageType || msg.Type == RetrieveMessageType || msg.Type == UpdateMessageType ||
 		msg.Type == DeleteMessageType {
-		return nil, fmt.Errorf("%s", "Invalid Source header string")
+		return nil, fmt.Errorf("%s", ErrInvalidSource)
 	}
 
 	// TransactionUuid is mandatory for SimpleRequestResponse and CRUD
-	if transUuid := header.Get(TrasactionUuidHeader); !strings.EqualFold(transUuid, "") {
+	if transUuid := header.Get(TransactionUuidHeader); !strings.EqualFold(transUuid, "") {
```
Comment (johnabass) [inline wrp/header_wrp.go]: Remove all validation from this code.  This block of code should only worry about translating headers into messages.  Let the `wrp` package handle validation (which already exists).

File: wrp/header_wrp.go (hunk 1 above)
Comment (shilpa24balaji) [inline wrp/header_wrp.go]: Ya makes sense I did as per this https://github.com/Comcast/webpa-common/issues/90

Comment (schmidtw) [inline wrp/header_wrp.go]: Not that I recall.

File: wrp/header_wrp.go (hunk 2 above)
Comment (schmidtw) [inline wrp/header_wrp.go]: Same as above since we could go from JSON -> WRP or HTTP Headers -> WRP and then the opposite.

**Pull request merged**